crates/ruff/resources/test/fixtures/flake8_boolean_trap/FBT.py

```diff
@@ -69,6 +69,8 @@ class Registry:
     def __init__(self) -> None:
         self._switches = [False] * len(Switch)
 
+        object.__setattr__(self, "flag", True)
```

Comment:
Python noob here. Is this a realistic example? Why not use `setattr(object, "flag", true)` or `object.flag = True`? If not, can we add a realistic example where the use of `__setattr__` over `setattr` is recommended?

Reply:
In that example, `object` would be the python base class for all classes, so basically
```
object.__setattr__(self, "flag", True)
```
will call the implementation of `setattr` of the `object` class, while
```
setattr(self, "flag", True)
```
is just syntactic sugar for `self.__setattr__(self, "flag", True)` and calls the implementation of the `self` class (here that's `Registry`). This is sometimes used when you want to bypass any custom user implementation of `__setattr__` and just call the default from `object`.

We can add another class here with an overwritten `__setattr__` to highlight the difference, but not sure it's worth it (?).

Reply:
For my real-life example, I used this pattern to memoize some properties on a frozen [dataclass](https://docs.python.org/3/library/dataclasses.html) and called `object.__setattr__(self, "cache_dict", {})`, which would be impossible with `setattr`.

Reply:
Thx, that helped